india_stocks_api/brokers/base.py

```
@@ -95,6 +100,29 @@ def _download_master_contract(self, db_path: str = 'instruments.db'):
 
+    def _require_auth(self) -> str:
+        """
+        Validate that a live, non-expired session exists.
+
+        Returns the JWT token if valid.
+        Raises AuthenticationError if not authenticated.
+        Raises SessionExpiredError if session is past midnight IST.
+        """
+        token = context.get_auth_token()
+        if not token:
+            raise AuthenticationError("Not authenticated. Call authenticate() first.")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)